app/jobs/update_glossary.rb

```diff
@@ -0,0 +1,30 @@
+
+class UpdateGlossary < ApplicationJob
+  queue_as :glossary
+
+  def perform
+    response = HTTParty.get("https://ssd.eff.org/en/glossary")
+    doc = Nokogiri::HTML(response.to_s)
+
+    doc.css(".view-glossary-list .views-row").each do |row|
+      name = row.css(".views-field-name-field a").text.strip
+      path = row.css(".views-field-name-field a").attr("href").value
+      body = row.css(".views-field-description-field").text.strip
+
+      response = HTTParty.get("https://ssd.eff.org#{path}")
+      doc = Nokogiri::HTML(response.to_s)
+      synonyms = doc.css(".field-name-field-synonym .field-item").
+                 map(&:text).map(&:strip)
+
+      body = body.lines.map{ |p| "<p>#{p.strip}</p>" }.join
+
+      unless GlossaryTerm.find_by(name: name)
```

Comment:
This might get pretty db-intensive, especially as the GlossaryTerm table grows, especially since the job is run four times an hour.  It means 1 db call per name, + 1 db call if the name is new, for a worst case of N*2 and a best case of N.  Would it be more efficient to do something like this?
```
name = []
terms = doc.css(".view-glossary-list .views-row").map do |row|
  body = row.css(".views-field-description-field").text.strip
  body = body.lines.map{ |p| "<p>#{p.strip}</p>" }.join,

  path = row.css(".views-field-name-field a").attr("href").value
  response = HTTParty.get("https://ssd.eff.org#{path}")
  doc = Nokogiri::HTML(response.to_s)
  synonyms = doc.css(".field-name-field-synonym .field-item").
    map(&:text).map(&:strip)

  name = row.css(".views-field-name-field a").text.strip
  names << name
  { name: name, body: body, synonyms: synonyms }
end

names = names - GlossaryTerms.pluck(:name)
names.each do |name|
  term = terms.find { |term| term[:name] == name }
  GlossaryTerm.create!(
    name: name,
    body: term[:body],
    synonyms: term[:synonym]
  )
end
```
Then it's 1 db call for the `pluck`, and another for each missing name, for a worst case of N + 1 and a best case of 1.

Reply:
Ok, updated.